In [1]:
import pandas as pd
import numpy as np
data_google_playstore = pd.read_csv("googleplaystore.csv", encoding='latin1')

In [4]:
data_google_playstore.shape

(10841, 13)

* * *
***Question 1***

In your Jupyter Notebook or Google Colab, import numpy and pandas library, then perform loading dataset that you have got and stored in a folder named dataset. Typical code for loading dataset is as follows: 

google_play_apps = pd.read_csv('dataset/googleplaystore.csv')

In the dataset of googleplaystore.csv, it consists of  blank rows and  blank columns. 


***The Answer of Question no 1***
<br>
10841 rows and 13 columns

***Solution***
<br>
data_google_playstore.shape
* * *


* * *
***Question 2***
<br>
How many unique apps categories reflected in the dataset? 
<br>
<br>
***The Answer of Question no 2***
<br>
34 Apps Categories

***Solution***
<br>
data_google_playstore.groupby('Category').nunique().shape
* * *

In [5]:
data_google_playstore.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hid...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [20]:
data_google_playstore.groupby('Category')['App'].nunique().shape

(34,)

In [12]:
data_google_playstore[data_google_playstore.Category=='1.9']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [21]:
data_google_playstore.groupby('Category').nunique()

App  Category  Rating  Reviews  Size  Installs  Type  \
Category                                                                     
1.9                     1         1       1        1     1         1     1   
ART_AND_DESIGN         64         1      14       64    48        12     2   
AUTO_AND_VEHICLES      85         1      18       72    57        15     2   
BEAUTY                 53         1      14       48    37        12     1   
BOOKS_AND_REFERENCE   222         1      20      168   105        17     2   
BUSINESS              420         1      31      205   139        18     2   
COMICS                 56         1      19       60    41        11     1   
COMMUNICATION         316         1      25      290   118        19     2   
DATING                171         1      27      153    81        15     2   
EDUCATION             119         1      13      130    57         9     2   
ENTERTAINMENT         102         1      16      110    42        10     2   
EVENTS                 64         1      14       50    45        12     2   
FAMILY               1909         1      35     1293   213        21     2   
FINANCE               345         1      33      287   117        17     2   
FOOD_AND_DRINK        112         1      21      110    52        15     2   
GAME                  960         1      28     1038   151        19     2   
HEALTH_AND_FITNESS    288         1      30      251   121        18     2   
HOUSE_AND_HOME         74         1      15       75    48        13     1   
LIBRARIES_AND_DEMO     84         1      17       79    72        12     2   
LIFESTYLE             369         1      33      258   135        18     2   
MAPS_AND_NAVIGATION   131         1      23      125    75        15     2   
MEDICAL               396         1      30      212   140        15     2   
NEWS_AND_MAGAZINES    254         1      27      216    86        18     2   
PARENTING              60         1      17       58    43        11     2   
PERSONALIZATION       376         1      22      276   140        18     2   
PHOTOGRAPHY           281         1      27      294   113        17     2   
PRODUCTIVITY          374         1      29      306   131        20     2   
SHOPPING              202         1      21      203    93        15     2   
SOCIAL                239         1      22      238   105        20     2   
SPORTS                327         1      26      282   132        17     2   
TOOLS                 829         1      37      592   204        19     2   
TRAVEL_AND_LOCAL      219         1      24      204    87        19     2   
VIDEO_PLAYERS         164         1      26      161    84        16     2   
WEATHER                79         1      15       79    45        12     2   

                     Price  Content Rating  Genres  Last Updated  Current Ver  \
Category                                                                        
1.9                      1               0       1             1            1   
ART_AND_DESIGN           2               3       4            52           41   
AUTO_AND_VEHICLES        4               3       1            55           64   
BEAUTY                   1               4       1            48           37   
BOOKS_AND_REFERENCE     14               4       1           180          113   
BUSINESS                11               4       1           263          254   
COMICS                   1               5       2            39           44   
COMMUNICATION           10               3       1           200          194   
DATING                   6               3       1            69          110   
EDUCATION                3               4       7            80           82   
ENTERTAINMENT            3               4       4            53           61   
EVENTS                   2               3       1            56           46   
FAMILY                  33               5      73           742          711  

In [22]:
data_google_playstore.groupby('Category').nunique().shape

(34, 13)

* * *
***Question 3***
<br>
How many unique genres according to the dataset? 
<br>
<br>
***The Answer of Question no 3***
<br>
120 Unique Apps Genres

***Solution***
<br>
data_google_playstore.groupby('Genres').nunique().shape
* * *

In [23]:
data_google_playstore.groupby('Genres').nunique().shape

(120, 13)

In [25]:
data_google_playstore.groupby('Genres').nunique()

App  Category  Rating  Reviews  Size  \
Genres                                                                        
Action                                 299         1      18      347   100   
Action;Action & Adventure               12         2       7       15    11   
Adventure                               73         1      14       75    53   
Adventure;Action & Adventure             5         2       3       12     5   
Adventure;Brain Games                    1         1       1        1     1   
...                                    ...       ...     ...      ...   ...   
Video Players & Editors                163         1      26      159    84   
Video Players & Editors;Creativity       1         2       1        2     1   
Video Players & Editors;Music & Video    2         2       1        3     2   
Weather                                 79         1      15       79    45   
Word                                    23         1      11       28    21   

                                       Installs  Type  Price  Content Rating  \
Genres                                                                         
Action                                       16     2      8               4   
Action;Action & Adventure                     5     2      4               2   
Adventure                                    13     2      5               4   
Adventure;Action & Adventure                  5     2      2               2   
Adventure;Brain Games                         1     1      1               1   
...                                         ...   ...    ...             ...   
Video Players & Editors                      16     2      5               4   
Video Players & Editors;Creativity            1     1      1               1   
Video Players & Editors;Music & Video         2     1      1               2   
Weather                                      12     2      7               4   
Word                                         10     1      1               4   

                                       Genres  Last Updated  Current Ver  \
Genres                                                                     
Action                                      1           198          175   
Action;Action & Adventure                   1            12           11   
Adventure                                   1            61           63   
Adventure;Action & Adventure                1             5            4   
Adventure;Brain Games                       1             1            1   
...                                       ...           ...          ...   
Video Players & Editors                     1           117          102   
Video Players & Editors;Creativity          1             1            1   
Video Players & Editors;Music & Video       1             2            2   
Weather                                     1            62           52   
Word                                        1            20           20   

                                       Android Ver  
Genres                                              
Action                                          19  
Action;Action & Adventure                        7  
Adventure                                       13  
Adventure;Action & Adventure                     2  
Adventure;Brain Games                            1  
...                                            ...  
Video Players & Editors                         16  
Video Players & Editors;Creativity               1  
Video Players & Editors;Music & Video            2  
Weather                                         14  
Word                                             9  

[120 rows x 13 columns]

* * *
***Question 4***
<br>
Which code is appropriate to drop duplicate values in 'App' column and remove its rows in Python?  Assume dataset was loaded as a pandas data frame variable: google_play_apps.
<br>
<br>
***The Answer of Question no 4***
<br>
google_play_apps.drop_duplicates(subset='App', inplace=False)

***Solution***
<br>
google_play_apps.drop_duplicates(subset='App', inplace=False)
* * *

In [45]:
data_google_playstore.groupby(['App']).size().reset_index(name='counts').sort_values(by=['counts'], ascending=False)

App  counts
7302                                             ROBLOX       9
1934  CBS Sports App - Scores, News, Stats & Watch Live       8
2364                                   Candy Crush Saga       7
3504                     Duolingo: Learn Languages Free       7
101                                         8 Ball Pool       7
...                                                 ...     ...
3310                         Display Phone Screen On TV       1
3311                                  Displaying You VR       1
3312                             Ditto CP Calculator GO       1
3313                                      Dive-Store.ch       1
9659  🔥 Football Wallpapers 4K | Full HD Backgrou...       1

[9660 rows x 2 columns]

In [57]:
data_google_playstore.drop_duplicates(subset='App', inplace=False).groupby(['App']).size().reset_index(name='counts').sort_values(by=['counts'], ascending=False)

App  counts
0                "i DT" Fútbol. Todos Somos Técnicos.       1
6434                                     My Talking Tom       1
6436                        My Teacher - Classroom Play       1
6437                                          My Telcel       1
6438                                           My Tele2       1
...                                                 ...     ...
3220  Delivery trough - delivery trough delivery trough       1
3221                                     Delivery yogi.       1
3222                                       Delta Dental       1
3223                             Denis Brogniart - AH !       1
9659  🔥 Football Wallpapers 4K | Full HD Backgrou...       1

[9660 rows x 2 columns]

In [58]:
data_google_playstore.drop_duplicates(subset='App', inplace=False)

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hid...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[9660 rows x 13 columns]

In [59]:
data_google_playstore[data_google_playstore.App=='ROBLOX']

App Category  Rating  Reviews Size      Installs  Type Price  \
1653  ROBLOX     GAME     4.5  4447388  67M  100,000,000+  Free     0   

     Content Rating                        Genres   Last Updated  \
1653   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   

       Current Ver Android Ver  
1653  2.347.225742  4.1 and up

In [60]:
data_google_playstore.groupby(['App']).size().reset_index(name='counts').sort_values(by=['counts'], ascending=False)

App  counts
0                "i DT" Fútbol. Todos Somos Técnicos.       1
6434                                     My Talking Tom       1
6436                        My Teacher - Classroom Play       1
6437                                          My Telcel       1
6438                                           My Tele2       1
...                                                 ...     ...
3220  Delivery trough - delivery trough delivery trough       1
3221                                     Delivery yogi.       1
3222                                       Delta Dental       1
3223                             Denis Brogniart - AH !       1
9659  🔥 Football Wallpapers 4K | Full HD Backgrou...       1

[9660 rows x 2 columns]

* * *
***Question 5***
<br>
Upon removing duplicate of App column, please run the following code to remove a row that contain 'Free' value at the 'Installs' column. You can check first with this code: 

google_play_apps[google_play_apps['Installs']=='Free']
To remove that specific row: 

google_play_apps = google_play_apps[google_play_apps['Installs'] != 'Free']
If you want to remove character '+' in column 'Installs', you can run the following codes in Python. 

google_play_apps['Installs'] = google_play_apps blank .apply(lambda x: x.replace('+', '') if '+' in str(x) else x)
Then, if you want to remove character ',' in column 'Installs', you can run the following codes in Python. 

google_play_apps['Installs'] =  blank ['Installs'].apply(lambda x: x.replace( blank , '') if ',' in str(x) else x)
Assume that you loaded dataset as a pandas data frame with variable name: google_play_apps. 

To understand lambda, please read this article.
<br>
<br>
***The Answer of Question no 5***
<br>
google_play_apps['Installs'] = google_play_apps['Installs'].apply(lambda x: x.replace('+', '') if '+' in str(x) else x)

google_play_apps['Installs'] = google_play_apps['Installs'].apply(lambda x: x.replace(',', '') if ',' in str(x) else x)

***Solution***
<br>
google_play_apps['Installs'] = google_play_apps['Installs'].apply(lambda x: x.replace('+', '') if '+' in str(x) else x)

google_play_apps['Installs'] = google_play_apps['Installs'].apply(lambda x: x.replace(',', '') if ',' in str(x) else x)
* * *

In [65]:
data_google_playstore.groupby('Installs').size()

Installs
0                    1
0+                  14
1+                  67
1,000+             888
1,000,000+        1417
1,000,000,000+      20
10+                385
10,000+           1031
10,000,000+        937
100+               710
100,000+          1112
100,000,000+       188
5+                  82
5,000+             468
5,000,000+         607
50+                204
50,000+            469
50,000,000+        202
500+               328
500,000+           505
500,000,000+        24
Free                 1
dtype: int64

In [66]:
data_google_playstore[data_google_playstore.Installs == 'Free']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [67]:
data_google_playstore = data_google_playstore[data_google_playstore.Installs != 'Free']

In [68]:
data_google_playstore[data_google_playstore.Installs == 'Free']

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [69]:
data_google_playstore.groupby('Installs').size()

Installs
0                    1
0+                  14
1+                  67
1,000+             888
1,000,000+        1417
1,000,000,000+      20
10+                385
10,000+           1031
10,000,000+        937
100+               710
100,000+          1112
100,000,000+       188
5+                  82
5,000+             468
5,000,000+         607
50+                204
50,000+            469
50,000,000+        202
500+               328
500,000+           505
500,000,000+        24
dtype: int64

In [70]:
data_google_playstore['Installs'].head()

0        10,000+
1       500,000+
2     5,000,000+
3    50,000,000+
4       100,000+
Name: Installs, dtype: object

In [72]:
data_google_playstore['Installs'] = data_google_playstore['Installs'].apply(lambda x: x.replace('+', '') if '+' in str(x) else x)

c:\users\permadiher\appdata\local\continuum\miniconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [74]:
data_google_playstore.groupby('Installs').size()

Installs
0                  15
1                  67
1,000             888
1,000,000        1417
1,000,000,000      20
10                385
10,000           1031
10,000,000        937
100               710
100,000          1112
100,000,000       188
5                  82
5,000             468
5,000,000         607
50                204
50,000            469
50,000,000        202
500               328
500,000           505
500,000,000        24
dtype: int64

In [89]:
data_google_playstore['Installs'] = data_google_playstore['Installs'].apply(lambda x: x.replace(',', '') if ',' in str(x) else x)

c:\users\permadiher\appdata\local\continuum\miniconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [90]:
data_google_playstore['Installs'].head()

0       10000
1      500000
2     5000000
3    50000000
4      100000
Name: Installs, dtype: object

* * *
***Question 6***
<br>
If you want to convert data type of 'Installs' columns into integer, you can run the following code. 

google_play_apps blank = google_play_apps blank .astype( blank )
Assume you loaded dataset as a pandas data frame with variable name: google_play_apps. 
<br>
<br>
***The Answer of Question no 6***
data_google_playstore['Installs'] = data_google_playstore['Installs'].astype(int)

***Solution***
<br>
data_google_playstore['Installs'] = data_google_playstore['Installs'].astype(int)
* * *

In [94]:
data_google_playstore.dtypes

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

In [95]:
data_google_playstore['Installs'] = data_google_playstore['Installs'].astype(int)

c:\users\permadiher\appdata\local\continuum\miniconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [96]:
data_google_playstore.dtypes

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs            int32
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

In [106]:
data_google_playstore[(data_google_playstore.App == 'DU Battery Saver - Battery Charger & Battery Life') 
                      | (data_google_playstore.App == 'Flipboard: News For Our Time')
                      | (data_google_playstore.App == 'Pinterest')
                      | (data_google_playstore.App == 'Dropbox')
                      | (data_google_playstore.App == 'Subway Surfers')
                      | (data_google_playstore.App == 'WhatsApp Messenger')
                      | (data_google_playstore.App == 'File Commander - File Manager/Explorer')
                    | (data_google_playstore.App == 'FIFA Soccer')
                     ]

App            Category  \
336                                   WhatsApp Messenger       COMMUNICATION   
1654                                      Subway Surfers                GAME   
2552                                           Pinterest              SOCIAL   
3473                                             Dropbox        PRODUCTIVITY   
3755                        Flipboard: News For Our Time  NEWS_AND_MAGAZINES   
8445                                         FIFA Soccer              SPORTS   
8896   DU Battery Saver - Battery Charger & Battery Life               TOOLS   
10253             File Commander - File Manager/Explorer            BUSINESS   

       Rating   Reviews                Size    Installs  Type Price  \
336       4.4  69119316  Varies with device  1000000000  Free     0   
1654      4.5  27722264                 76M  1000000000  Free     0   
2552      4.6   4305441  Varies with device   100000000  Free     0   
3473      4.4   1861310                 61M   500000000  Free     0   
3755      4.4   1284017  Varies with device   500000000  Free     0   
8445      4.2   3909032                 51M   100000000  Free     0   
8896      4.5  13479633                 14M   100000000  Free     0   
10253     4.3    758590                 12M   100000000  Free     0   

      Content Rating            Genres    Last Updated         Current Ver  \
336         Everyone     Communication  August 3, 2018  Varies with device   
1654    Everyone 10+            Arcade   July 12, 2018              1.90.0   
2552            Teen            Social  August 3, 2018  Varies with device   
3473        Everyone      Productivity  August 1, 2018  Varies with device   
3755    Everyone 10+  News & Magazines  August 3, 2018  Varies with device   
8445        Everyone            Sports   July 31, 2018             10.5.00   
8896        Everyone             Tools    June 5, 2018             4.8.7.8   
10253       Everyone          Business  August 2, 2018  Varies with device   

              Android Ver  
336    Varies with device  
1654           4.1 and up  
2552   Varies with device  
3473   Varies with device  
3755   Varies with device  
8445           4.1 and up  
8896           4.0 and up  
10253          4.1 and up

* * *
***Question 7***
<br>
Match Apps and Category based on provided dataset! 

(Hint: Use App and Category columns)

DU Battery Saver - Battery Charger & Battery Life

Answer 1
Flipboard: News For Our Time	Answer 2
Pinterest

Answer 3
Dropbox

Answer 4
Subway Surfers

Answer 5
WhatsApp Messenger

Answer 6
File Commander - File Manager/Explorer

Answer 7
FIFA Soccer
<br>
<br>
***The Answer of Question no 7***



DU Battery Saver - Battery Charger & Battery Life ***TOOLS***
<br>
Flipboard: News For Our Time	***NEWS_AND_MAGAZINES***
<br>
Pinterest ***SOCIAL***
<br>
Dropbox ***PRODUCTIVITY***
<br>
Subway Surfers ***GAME***
<br>
WhatsApp Messenger ***COMMUNICATION***
<br>
File Commander - File Manager/Explorer ***BUSINESS***
<br>
FIFA Soccer ***SPORT***
<br>


***Solution***
<br>
data_google_playstore[(data_google_playstore.App == 'DU Battery Saver - Battery Charger & Battery Life') 
                      | (data_google_playstore.App == 'Flipboard: News For Our Time')
                      | (data_google_playstore.App == 'Pinterest')
                      | (data_google_playstore.App == 'Dropbox')
                      | (data_google_playstore.App == 'Subway Surfers')
                      | (data_google_playstore.App == 'WhatsApp Messenger')
                      | (data_google_playstore.App == 'File Commander - File Manager/Explorer')
                    | (data_google_playstore.App == 'FIFA Soccer')
                     ]
* * *

* * *
***Question 8***
<br>
The following applications were installed by more than 1 billion users, EXCEPT ... . 

Select one:
a. Facebook
b. WhatsApp Messenger
c. Google Duo - High Quality Video Calls 
d. Skype

***The Answer of Question no 8***
<br>
Google Duo - High Quality Video Calls 
<br>

***Solution***
<br>
data_google_playstore[data_google_playstore.Installs >= 1000000000]

* * *

In [107]:
data_google_playstore.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hid...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size  Installs  Type Price Content Rating  \
0     159   19M     10000  Free     0       Everyone   
1     967   14M    500000  Free     0       Everyone   
2   87510  8.7M   5000000  Free     0       Everyone   
3  215644   25M  50000000  Free     0           Teen   
4     967  2.8M    100000  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [112]:
data_google_playstore[data_google_playstore.Installs >= 1000000000]

App             Category  Rating  \
152                            Google Play Books  BOOKS_AND_REFERENCE     3.9   
335   Messenger – Text and Video Chat for Free        COMMUNICATION     4.0   
336                           WhatsApp Messenger        COMMUNICATION     4.4   
338                 Google Chrome: Fast & Secure        COMMUNICATION     4.3   
340                                        Gmail        COMMUNICATION     4.3   
341                                     Hangouts        COMMUNICATION     4.0   
391                Skype - free IM & video calls        COMMUNICATION     4.1   
865                            Google Play Games        ENTERTAINMENT     4.3   
1654                              Subway Surfers                 GAME     4.5   
2544                                    Facebook               SOCIAL     4.1   
2545                                   Instagram               SOCIAL     4.5   
2554                                     Google+               SOCIAL     4.2   
2808                               Google Photos          PHOTOGRAPHY     4.5   
3117                   Maps - Navigate & Explore     TRAVEL_AND_LOCAL     4.3   
3127                          Google Street View     TRAVEL_AND_LOCAL     4.2   
3234                                      Google                TOOLS     4.4   
3454                                Google Drive         PRODUCTIVITY     4.4   
3665                                     YouTube        VIDEO_PLAYERS     4.3   
3687                     Google Play Movies & TV        VIDEO_PLAYERS     3.7   
3736                                 Google News   NEWS_AND_MAGAZINES     3.9   

       Reviews                Size    Installs  Type Price Content Rating  \
152    1433233  Varies with device  1000000000  Free     0           Teen   
335   56642847  Varies with device  1000000000  Free     0       Everyone   
336   69119316  Varies with device  1000000000  Free     0       Everyone   
338    9642995  Varies with device  1000000000  Free     0       Everyone   
340    4604324  Varies with device  1000000000  Free     0       Everyone   
341    3419249  Varies with device  1000000000  Free     0       Everyone   
391   10484169  Varies with device  1000000000  Free     0       Everyone   
865    7165362  Varies with device  1000000000  Free     0           Teen   
1654  27722264                 76M  1000000000  Free     0   Everyone 10+   
2544  78158306  Varies with device  1000000000  Free     0           Teen   
2545  66577313  Varies with device  1000000000  Free     0           Teen   
2554   4831125  Varies with device  1000000000  Free     0           Teen   
2808  10858556  Varies with device  1000000000  Free     0       Everyone   
3117   9235155  Varies with device  1000000000  Free     0       Everyone   
3127   2129689  Varies with device  1000000000  Free     0       Everyone   
3234   8033493  Varies with device  1000000000  Free     0       Everyone   
3454   2731171  Varies with device  1000000000  Free     0       Everyone   
3665  25655305  Varies with device  1000000000  Free     0           Teen   
3687    906384  Varies with device  1000000000  Free     0           Teen   
3736    877635                 13M  1000000000  Free     0           Teen   

                       Genres    Last Updated         Current Ver  \
152         Books & Reference  August 3, 2018  Varies with device   
335             Communication  August 1, 2018  Varies with device   
336             Communication  August 3, 2018  Varies with device   
338             Communication  August 1, 2018  Varies with device   
340             Communication  August 2, 2018  Varies with device   
341             Communication   July 21, 2018  Varies with device   
391             Communication  August 3, 2018  Varies with device   
865             Entertainment   July 16, 2018  Varies with device   
1654                   Arcade   July 12, 2018              1.90.0   
2544                   Social  Augu

In [114]:
data_google_playstore.dtypes

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs            int32
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

In [115]:
data_google_playstore['Reviews'] = data_google_playstore['Reviews'].astype(int)

c:\users\permadiher\appdata\local\continuum\miniconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [116]:
data_google_playstore.dtypes

App                object
Category           object
Rating            float64
Reviews             int32
Size               object
Installs            int32
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

In [121]:
data_google_playstore[data_google_playstore.Category=='COMMUNICATION'].sort_values(by=['Reviews'], ascending=False).head(50)

App       Category  \
336                                  WhatsApp Messenger  COMMUNICATION   
335          Messenger – Text and Video Chat for Free  COMMUNICATION   
378         UC Browser - Fast Download Private & Secure  COMMUNICATION   
406                         BBM - Free Calls & Messages  COMMUNICATION   
342                                     Viber Messenger  COMMUNICATION   
403                         LINE: Free Calls & Messages  COMMUNICATION   
391                       Skype - free IM & video calls  COMMUNICATION   
338                        Google Chrome: Fast & Secure  COMMUNICATION   
449   Truecaller: Caller ID, SMS spam blocking & Dialer  COMMUNICATION   
365                                              WeChat  COMMUNICATION   
351                       Opera Mini - fast web browser  COMMUNICATION   
347                       imo free video calls and chat  COMMUNICATION   
340                                               Gmail  COMMUNICATION   
345                       Yahoo Mail – Stay Organized  COMMUNICATION   
366         UC Browser Mini -Tiny Fast Private & Secure  COMMUNICATION   
341                                            Hangouts  COMMUNICATION   
370                                            Telegram  COMMUNICATION   
344                      Firefox Browser fast & private  COMMUNICATION   
395          GO SMS Pro - Messenger, Free Themes, Emoji  COMMUNICATION   
407                        KakaoTalk: Free Calls & Text  COMMUNICATION   
418       Dolphin Browser - Fast, Private & Adblock🐬  COMMUNICATION   
356                      Opera Browser: Fast and Secure  COMMUNICATION   
4279                                                Kik  COMMUNICATION   
363                                                 Who  COMMUNICATION   
413   CM Browser - Ad Blocker , Fast Download , Privacy  COMMUNICATION   
371               Google Duo - High Quality Video Calls  COMMUNICATION   
339               Messenger Lite: Free Calls & Messages  COMMUNICATION   
424                      DU Browser—Browse fast & fun  COMMUNICATION   
4739  WhatsCall Free Global Phone Call App & Cheap C...  COMMUNICATION   
3892                                               Azar  COMMUNICATION   
6297                                Zalo – Video Call  COMMUNICATION   
459                                 Mail.Ru - Email App  COMMUNICATION   
369                                    Android Messages  COMMUNICATION   
4518               ICQ — Video Calls & Chat Messenger  COMMUNICATION   
346                        imo beta free calls and text  COMMUNICATION   
362    TouchPal Keyboard - Fun Emoji & Android Keyboard  COMMUNICATION   
389                           free video calls and chat  COMMUNICATION   
441                        Whoscall - Caller ID & Block  COMMUNICATION   
372                                  Puffin Web Browser  COMMUNICATION   
440   CallApp: Caller ID, Blocker & Phone Call Recorder  COMMUNICATION   
443                      Calls Blacklist - Call Blocker  COMMUNICATION   
354                             ZenUI Dialer & Contacts  COMMUNICATION   
4426                            Opera Mini browser beta  COMMUNICATION   
399                        Glide - Video Chat Messenger  COMMUNICATION   
400                                            Text SMS  COMMUNICATION   
401                                         Google Allo  COMMUNICATION   
398                                           chomp SMS  COMMUNICATION   
404                                             GroupMe  COMMUNICATION   
4033                          Discord - Chat for Gamers  COMMUNICATION   
454   myMail – Email for Hotmail, Gmail and Outloo...  COMMUNICATION   

      Rating   Reviews                Size    Installs  Type Price  \
336      4.4  69119316  Varies with device  1000000000  Free     0   
335      4.0  56642847  Varies with device  1000000000  Free     0   
378      4.5  17712922                 40M   500000000  Free     0 

* * *
***Question 9***
<br>
Arrange top COMMUNICATION category apps from the highest number of reviews to the lowest! 

(Hint: Convert the data type to integer first. It is similar with code that was asked in question 6. 

***The Answer of Question no 9***
<br>
1. Whats app messenger
2. BBM
3. Viber Messenger
4. Line
5. Skype
<br>

***Solution***
<br>
data_google_playstore['Reviews'] = data_google_playstore['Reviews'].astype(int)
data_google_playstore[data_google_playstore.Category=='COMMUNICATION'].sort_values(by=['Reviews'], ascending=False).head(50)

* * * 

In [122]:
data_google_playstore['Rating'] = data_google_playstore['Rating'].astype(float)

c:\users\permadiher\appdata\local\continuum\miniconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [123]:
data_google_playstore.dtypes

App                object
Category           object
Rating            float64
Reviews             int32
Size               object
Installs            int32
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

In [125]:
data_google_playstore[data_google_playstore.Category == 'GAME'].sort_values(by=['Installs','Rating'], ascending=False)

App Category  Rating   Reviews                Size  \
1654      Subway Surfers     GAME     4.5  27722264                 76M   
1722      My Talking Tom     GAME     4.5  14891223  Varies with device   
1655    Candy Crush Saga     GAME     4.4  22426677                 74M   
1661        Temple Run 2     GAME     4.3   8118609                 62M   
1662                 Pou     GAME     4.3  10485308                 24M   
...                  ...      ...     ...       ...                 ...   
5917            Ra Ga Ba     GAME     5.0         2                 20M   
10697            Mu.F.O.     GAME     5.0         2                 16M   
4218   D+H Reaction Wall     GAME     NaN         0  Varies with device   
5846           YAKALA AY     GAME     NaN         0                 14M   
6277    Bi-Tank Ads Free     GAME     NaN         0  Varies with device   

         Installs  Type  Price Content Rating  Genres      Last Updated  \
1654   1000000000  Free      0   Everyone 10+  Arcade     July 12, 2018   
1722    500000000  Free      0       Everyone  Casual     July 19, 2018   
1655    500000000  Free      0       Everyone  Casual      July 5, 2018   
1661    500000000  Free      0       Everyone  Action      July 5, 2018   
1662    500000000  Free      0       Everyone  Casual      May 25, 2018   
...           ...   ...    ...            ...     ...               ...   
5917            1  Paid  $1.49       Everyone  Arcade  February 8, 2017   
10697           1  Paid  $0.99       Everyone  Arcade     March 3, 2017   
4218            1  Paid  $0.99       Everyone  Arcade     July 24, 2016   
5846            1  Paid  $0.99       Everyone  Arcade      July 7, 2018   
6277            1  Paid  $0.99       Everyone  Arcade     July 31, 2017   

              Current Ver         Android Ver  
1654               1.90.0          4.1 and up  
1722            4.8.0.132          4.1 and up  
1655            1.129.0.2          4.1 and up  
1661               1.49.1          4.0 and up  
1662               1.4.77          4.0 and up  
...                   ...                 ...  
5917                1.0.4          2.3 and up  
10697                 1.0          2.3 and up  
4218   Varies with device  Varies with device  
5846                  1.0          4.1 and up  
6277                  1.1          4.1 and up  

[959 rows x 13 columns]

* * *
***Question 10***
<br>
Arrange top 3 game category apps based on number of installed and rating!

***The Answer of Question no 10***
<br>
1. Subway Surfers
2. My Talking Tom
3. Candy Crush Saga
<br>

***Solution***
<br>
data_google_playstore['Rating'] = data_google_playstore['Rating'].astype(float)
data_google_playstore[data_google_playstore.Category == 'GAME'].sort_values(by=['Installs','Rating'], ascending=False)

* * * 